include/eve/algo/ptr_iterator.hpp

```diff
@@ -38,7 +38,7 @@ namespace eve::algo
     template <typename _Cardinal>
     auto cardinal_cast(_Cardinal) const { return unaligned_ptr_iterator<T, _Cardinal>{ptr}; }
 
-    value_type& operator*() const { return *ptr; }
+    auto& operator*() const { return *ptr; }
```

Comment:
why this change ?

Reply:
Because `value_type` for `const int*` is `int` but the reference type is `const int&`